Modules/Brand/Http/Services/BrandService.php

```diff
@@ -0,0 +1,509 @@
+                    $file->move($brandAbsPath, $fileName);
+                    $catalog = new Catalog();
+                    $catalog->brand_id = $brand->id;
+                    $catalog->filename = $brandRelPath . $fileName;
+                    $catalog->save();
+                }
+            }
+            if ($request->file('upload_zip')) {
+
+                $fileName = Str::random(10) . '_photos.' . $request->file('upload_zip')->extension();
+                $request->file('upload_zip')->move($brandAbsPath, $fileName);
+                $brand->upload_zip = $brandRelPath . $fileName;
+                $brand->save();
+            }
+            if ($request->file('upload_contact_list')) {
+                $fileName = Str::random(10) . '_cstmrs.' . $request->file('upload_contact_list')->extension();
+                $request->file('upload_contact_list')->move($brandAbsPath, $fileName);
+                $brand->upload_contact_list = $brandRelPath . $fileName;
+                $brand->save();
+            }
+            $response = ['res' => true, 'msg' => "", 'data' => $data];
+        } catch (Exception $e) {
+            $errorCode = $e->getCode();
+            if ($errorCode == 23000) {
+                $errorMessage = "direct link already exists";
+            } else {
+                $errorMessage = $e->getMessage();//"something went wrong, please try again";
+            }
+            $response = ['res' => false, 'msg' => $errorMessage, 'data' => $errorCode];
+
+        }
+        return $response;
+    }
+
+    /**
+     * Save image from base64 string.
+     *
+     * @param int $brand
+     * @param $image
+     * @param $previousFile
+     * @param $replaceable
+     * @return Stringable
+     */
+    private function imageUpload(int $brand, $image, $previousFile, $replaceable): Stringable|string
+    {
+
+        $brandAbsPath = $this->brandAbsPath . '/' . $brand . '/';
+        $brandRelPath = $this->brandRelPath . $brand . '/';
+
+        if (!file_exists($brandAbsPath)) {
+            mkdir($brandAbsPath, 0777, true);
+        }
+
+        if ($replaceable && $previousFile !== null) {
+            $unlinkUrl = public_path() . $previousFile;
+            if (file_exists($unlinkUrl)) {
+                unlink($unlinkUrl);
+            }
+        }
+
+        $image_64 = $image; //your base64 encoded data
+        $extension = explode('/', explode(':', substr($image_64, 0, strpos($image_64, ';')))[1])[1];   // .jpg .png .pdf
+        $replace = substr($image_64, 0, strpos($image_64, ',') + 1);
+        $image_64 = str_replace($replace, '', $image_64);
+        $image_64 = str_replace(' ', '+', $image_64);
+        $imageName = Str::random(10) . '.' . 'png';
+        File::put($brandAbsPath . $imageName, base64_decode($image_64));
+        return $brandRelPath . $imageName;
+    }
+
+    public function updateAccount($request): array
+    {
+        $validator = Validator::make($request->all(), [
```

Comment:
Validation need to be done in request
Authorization is missing

Reply:
done